app/views/dashboard/_diaper_drive.html.erb

```diff
@@ -1,7 +1,7 @@
 <div class="row">
   <div class="pull-left">
     <%= link_to donation do %>
-      <%= donation.line_items.total %> from <%= donation.diaper_drive_participant.business_name %>
+      <%= donation.line_items.total %> from <%= donation.diaper_drive ? donation.diaper_drive.name : donation.diaper_drive_participant.business_name %>
```

Comment:
`donation.diaper_drive&.name || donation.diaper_drive_participant&.business_name` might be a bit cleaner of a way to write this.